auth/src/main/java/com/firebase/ui/auth/ui/email/RecoverPasswordActivity.java

```diff
@@ -54,51 +60,69 @@ protected void onCreate(Bundle savedInstanceState) {
+                    } else {
+                        mEmailInputLayout.setError(state.getException().getLocalizedMessage());
+                    }
+                } else {
+                    getDialogHolder().showLoadingDialog(R.string.fui_progress_dialog_sending);
+                }
+            }
+        });
+
+        mEmailInputLayout = findViewById(R.id.email_layout);
         mEmailEditText = findViewById(R.id.email);
+        mEmailFieldValidator = new EmailFieldValidator(mEmailInputLayout);
 
         String email = getIntent().getStringExtra(ExtraConstants.EXTRA_EMAIL);
         if (email != null) {
             mEmailEditText.setText(email);
         }
 
+        ImeHelper.setImeOnDoneListener(mEmailEditText, this);
         findViewById(R.id.button_done).setOnClickListener(this);
     }
 
-    private void next(final String email) {
-        getAuthHelper().getFirebaseAuth()
-                .sendPasswordResetEmail(email)
-                .addOnFailureListener(
-                        new TaskFailureLogger(TAG, "Error sending password reset email"))
-                .addOnSuccessListener(new OnSuccessListener<Void>() {
-                    @Override
-                    public void onSuccess(Void aVoid) {
-                        getDialogHolder().dismissDialog();
-                        RecoveryEmailSentDialog.show(
-                                email, getSupportFragmentManager());
-                    }
-                })
-                .addOnFailureListener(this, new OnFailureListener() {
-                    @Override
-                    public void onFailure(@NonNull Exception e) {
-                        getDialogHolder().dismissDialog();
-
-                        if (e instanceof FirebaseAuthInvalidUserException) {
-                            // No FirebaseUser exists with this email address, show error.
-                            mEmailEditText.setError(getString(R.string.fui_error_email_does_not_exist));
-                        }
-                    }
-                });
-    }
-
     @Override
     public void onClick(View view) {
-        if (view.getId() == R.id.button_done) {
-            if (mEmailFieldValidator.validate(mEmailEditText.getText())) {
-                getDialogHolder().showLoadingDialog(R.string.fui_progress_dialog_sending);
-                next(mEmailEditText.getText().toString());
-            }
+        if (view.getId() == R.id.button_done
+                && mEmailFieldValidator.validate(mEmailEditText.getText())) {
+            onDonePressed();
         }
     }
+
+    @Override
+    public void onDonePressed() {
+        mHandler.startReset(mEmailEditText.getText().toString());
+    }
+
+    private void showEmailSentDialog(String email) {
+        new AlertDialog.Builder(this)
+                .setTitle(R.string.fui_title_confirm_recover_password)
+                .setMessage(getString(R.string.fui_confirm_recovery_body, email))
+                .setOnDismissListener(this)
```

Comment:
nit: would it be clearer just to define the onDismissListener inline here?

Reply:
Yep, done.